ui/src/app/views/cohort-edit/component.ts

```diff
@@ -3,52 +3,71 @@ import {Router, ActivatedRoute} from '@angular/router';
 
 import {Cohort} from 'generated';
 import {CohortsService} from 'generated';
-import {User} from 'app/models/user';
-import {UserService} from 'app/services/user.service';
-import {CohortEditService} from 'app/services/cohort-edit.service';
+import {WorkspaceComponent} from 'app/views/workspace/component';
 
 @Component({
   styleUrls: ['./component.css'],
   templateUrl: './component.html',
 })
 export class CohortEditComponent implements OnInit {
-  user: User;
   cohort: Cohort = {id: '', name: '', description: '', criteria: '', type: ''};
   cohortId: string;
   adding = false;
   constructor(
       private router: Router,
       private route: ActivatedRoute,
-      private userService: UserService,
       private cohortsService: CohortsService,
-      private CohortEditService: CohortEditService
   ) {}
 
   ngOnInit(): void {
-    this.userService.getLoggedInUser()
-        .then(user => this.user = user);
     this.cohortId = this.route.snapshot.url[4].path;
     if (this.route.snapshot.url[5] === undefined) {
       this.adding = true;
     } else {
-      this.CohortEditService.get(this.cohortId).then(cohort => {
-        this.cohort = cohort;
-      }).catch(error => {
-        alert('All of Us Researcher Workbench does not currently \
-support editing cohorts from the API.');
-        this.router.navigate(['../../..'], {relativeTo : this.route});
-      });
+      this.cohortsService
+          .getCohort(
+              WorkspaceComponent.DEFAULT_WORKSPACE_NS,
+              WorkspaceComponent.DEFAULT_WORKSPACE_ID,
+              this.cohortId)
+          .retry(2)
+          .subscribe(this.handleCohortFetched);
     }
   }
 
+  /**
+   * Copy in only the client-used fields of the Cohort.
+   * TODO: Remove this and copy the Cohort wholesale once deserialization of all fields
+   * works server-side. (Currently creationTime deserialization causes errors.)
+   */
+  handleCohortFetched = (cohort: Cohort) => {
```

Comment:
I wasn't familiar with declaring class members as arrow functions. It seems like the preferred alternative to this.handleCohortFeched.bind(this). Tradeoffs / explanation here: https://github.com/Microsoft/TypeScript/wiki/'this'-in-TypeScript .